### PR TITLE
search parameter/map_zoom: Test for null to allow 0 as value

### PR DESCRIPTION
**Pull request by pfirpfel**
Addresses @fredj [feedback](https://github.com/camptocamp/ngeo/pull/3001#discussion_r145355559).

### CODE REVIEW

File: contribs/gmf/src/directives/search.js [hunk 1]
```diff
@@ -965,7 +965,7 @@ gmf.SearchController.prototype.fulltextsearch_ = function(query, resultIndex, op
         const feature = format.readFeature(data.features[resultIndex - 1]);
         this.featureOverlay_.addFeature(feature);
         const fitOptions = /** @type {olx.view.FitOptions} */ ({});
-        if (opt_zoom) {
+        if (opt_zoom !== null) {
```
Comment (fredj) [inline contribs/gmf/src/directives/search.js]: if `opt_zoom` is not passed to the function, it will be `undefined` not `null`

Comment (fredj) [reply contribs/gmf/src/directives/search.js]: And the `@param` for `opt_zoom` should be changed to:
```
@param {number=} opt_zoom Optional zoom level.
```

See https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#optional-parameter-in-a-param-annotation

File: contribs/gmf/src/directives/search.js [hunk 2]
```diff
@@ -965,7 +965,7 @@ gmf.SearchController.prototype.fulltextsearch_ = function(query, resultIndex, op
         const feature = format.readFeature(data.features[resultIndex - 1]);
         this.featureOverlay_.addFeature(feature);
         const fitOptions = /** @type {olx.view.FitOptions} */ ({});
-        if (opt_zoom) {
+        if (typeof opt_zoom !== 'undefined') {
```
Comment (fredj) [inline contribs/gmf/src/directives/search.js]: can be simplified to `if (opt_zoom !== undefined) {`

Comment (sbrunner) [reply contribs/gmf/src/directives/search.js]: +1

**Comment by fredj** [comment]
travis fails (probably) because the branch is from your fork, not the camptocamp one

**Pull request merged**